core/src/View/MDSpan/Kokkos_MDSpan_Accessor.hpp

```diff
@@ -34,35 +34,30 @@ namespace Impl {
 
 template <class MemorySpace, class NestedAccessor>
 struct SpaceAwareAccessor {
+  // Part of Accessor Requirements
   using element_type     = typename NestedAccessor::element_type;
   using reference        = typename NestedAccessor::reference;
   using data_handle_type = typename NestedAccessor::data_handle_type;
-
   using offset_policy =
       SpaceAwareAccessor<MemorySpace, typename NestedAccessor::offset_policy>;
 
-  using memory_space = MemorySpace;
+  // Specific to SpaceAwareAccessor
+  using memory_space         = MemorySpace;
+  using nested_accessor_type = NestedAccessor;
 
   static_assert(is_memory_space_v<memory_space>);
 
   KOKKOS_DEFAULTED_FUNCTION
   constexpr SpaceAwareAccessor() = default;
 
-  template <class OtherNestedAccessorType,
-            std::enable_if_t<std::is_constructible_v<NestedAccessor,
-                                                     OtherNestedAccessorType>,
-                             int> = 0>
-  KOKKOS_FUNCTION constexpr SpaceAwareAccessor(
-      const SpaceAwareAccessor<MemorySpace, OtherNestedAccessorType>&
-          other) noexcept
-      : nested_acc(other.nested_acc) {}
-
-  template <class OtherNestedAccessorType,
-            std::enable_if_t<std::is_constructible_v<NestedAccessor,
-                                                     OtherNestedAccessorType>,
-                             int> = 0>
+  template <
+      class OtherMemorySpace, class OtherNestedAccessorType,
+      std::enable_if_t<
+          MemorySpaceAccess<MemorySpace, OtherMemorySpace>::assignable &&
+              std::is_constructible_v<NestedAccessor, OtherNestedAccessorType>,
+          int> = 0>
```

Comment:
This change does not seem to belong to this PR which the description advertise as being about fixing the return type of the `SpaceAwareAccessor::offset` member function.
Please withdraw it or make the case that it belongs here and update the title and description.

Reply:
I updated title and description